lua/harpoon/init.lua

```diff
@@ -171,7 +170,12 @@ function M.save()
     M.refresh_projects_b4update()
 
     log.trace("save(): Saving cache config to", cache_config)
-    Path:new(cache_config):write(vim.fn.json_encode(HarpoonConfig), "w")
+
+    local to_save = utils.deepcopy(HarpoonConfig)
+    if type(to_save.menu.width) == "function" then
+        to_save.menu.width = nil
+    end
+    Path:new(cache_config):write(vim.fn.json_encode(to_save), "w")
```

Comment:
A deep copy is not required. This can be achieved with a local temp variable. 

```suggestion
    local menu = HarpoonConfig.menu
    HarpoonConfig.menu = nil
    Path:new(cache_config):write(vim.fn.json_encode(HarpoonConfig), "w")
    HarpoonConfig.menu = menu
```